magento/magento2
# Add input-file argument to varnish:vcl:generate

This pull request allows us to pass in a path to a custom modified VCL file relative to the Magento root. This way, we can add our own custom VCL to our Git repo and generate it on deploy using a default Magento command.




### Manual testing scenarios (*)

1. Install clean Magento
2. Copy the core Varnish file into `app/etc`
3. Modify the file. For testing purposes, we append `# TEST`
4. Generate the VCL using the new `input-file` argument
5. The generated VCL is shown in stdout, and you should see `# TEST` at the end of the output of the previous command.

Copy & paste;

```
cp vendor/magento/module-page-cache/etc/varnish6.vcl app/etc/varnish6.vcl
echo "# TEST" >> app/etc/varnish6.vcl
bin/magento varnish:vcl:generate --input-file=app/etc/varnish6.vcl --export-version=6
```

### Questions or comments


If someone could tell me for which scenarios to write a unit test, I'll try to do my best.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#37363: Add input-file argument to varnish:vcl:generate
